packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -94,6 +94,54 @@ export enum EcsMetric {
    * GPUReservation Metric
    */
   GPU_RESERVATION = 'GPUReservation',
+  /**
+   * ActiveConnectionCount Metric
+   */
+  ACTIVE_CONNECTION_COUNT = 'ActiveConnectionCount',
+  /**
+   * NewConnectionCount Metric
+   */
+  NEW_CONNECTION_COUNT = 'NewConnectionCount',
+  /**
+   * ProcessedBytes Metric
+   */
+  PROCESSED_BYTES = 'ProcessedBytes',
+  /**
+   * RequestCount Metric
+   */
+  REQUEST_COUNT = 'RequestCount',
+  /**
+   * GrpcRequestCount Metric
+   */
+  GRPC_REQUEST_COUNT = 'GrpcRequestCount',
+  /**
+   * HTTPCode_Target_2XX_Count Metric
+   */
+  HTTP_CODE_TARGET_2XX_COUNT = 'HTTPCode_Target_2XX_Count',
+  /**
+   * HTTPCode_Target_3XX_Count Metric
+   */
+  HTTP_CODE_TARGET_3XX_COUNT = 'HTTPCode_Target_3XX_Count',
+  /**
+   * HTTPCode_Target_4XX_Count Metric
+   */
+  HTTP_CODE_TARGET_4XX_COUNT = 'HTTPCode_Target_4XX_Count',
+  /**
+   * HTTPCode_Target_5XX_Count Metric
+   */
+  HTTP_CODE_TARGET_5XX_COUNT = 'HTTPCode_Target_5XX_Count',
+  /**
+   * RequestCountPerTarget Metric
+   */
+  REQUEST_COUNT_PER_TARGET = 'RequestCountPerTarget',
+  /**
+   * TargetProcessedBytes Metric
+   */
+  TARGET_PROCESSED_BYTES = 'TargetProcessedBytes',
+  /**
+   * TargetResponseTime Metric
+   */
+  TARGET_REESPONSE_TIME = 'TargetResponseTime',
```

Comment:
Please try to find a pattern for this enum which will make SC metric validation easier and more future proof. Perhaps if we use an [enum like class](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#enums) with static methods we can store additional context on whether the given metric is a SC metric or not. @iamhopaul123 what do you think? Typescript does not have multiple inheritance, would make this trivial.

Reply:
I agree it'll make the validation easier, but my worry is still when more metrics are added, we would need to update the class before they could use it (and I'm not sure if the current list is a full list or there are still metrics that ECS accepts not listed here).

If we make it a list of metric names, we could check if the metric is one of the element when they don't have sc enabled and we just error out.